### PR TITLE
[CNI] zap logger telemetry package

### PR DESCRIPTION
**Pull request by paulyufan2**
**Reason for Change**:

This PR is to migrate zap logger for telemetry package

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

File: telemetry/telemetry.go [hunk 1]
```diff
@@ -7,12 +7,16 @@ import (
...
 	"github.com/Azure/azure-container-networking/aitelemetry"
+	zapLog "github.com/Azure/azure-container-networking/cni/log"
 	"github.com/Azure/azure-container-networking/common"
 	"github.com/Azure/azure-container-networking/log"
 	"github.com/Azure/azure-container-networking/platform"
 	"github.com/pkg/errors"
+	"go.uber.org/zap"
 )
 
+var logger = zapLog.CNILogger.With(zap.String("component", "cni-telemetry"))
```
Comment (tamilmani1989) [inline telemetry/telemetry.go:18]: Isn't logger passed as argument to NewTelemetryBuffer ? where this logger is used?

File: telemetry/telemetry.go [hunk 2]
```diff
@@ -157,7 +166,7 @@ func SendCNIEvent(tb *TelemetryBuffer, report *CNIReport) {
 		reportBytes, err := reportMgr.ReportToBytes()
 		if err == nil {
 			if _, err = tb.Write(reportBytes); err != nil {
-				log.Printf("Error writing to telemetry socket:%v", err)
+				logger.Error("Error writing to telemetry socket", zap.Error(err))
```
Comment (tamilmani1989) [inline telemetry/telemetry.go:169]: should use tb.Logger

File: telemetry/telemetrybuffer.go [hunk 3]
```diff
@@ -109,7 +120,11 @@ func (tb *TelemetryBuffer) StartServer() error {
 							var tmp map[string]interface{}
 							err = json.Unmarshal(reportStr, &tmp)
 							if err != nil {
-								log.Logf("StartServer: unmarshal error:%v", err)
+								if tb.logger != nil {
+									tb.logger.Error("StartServer: unmarshal error", zap.Error(err))
```
Comment (tamilmani1989) [inline telemetry/telemetrybuffer.go:124]: tab space after if?

**Review by tamilmani1989 — CHANGES_REQUESTED** [review]

File: telemetry/telemetry.go [hunk 4]
```diff
@@ -152,7 +149,7 @@ func SendCNIMetric(cniMetric *AIMetric, tb *TelemetryBuffer) error {
 		report, err = reportMgr.ReportToBytes()
 		if err == nil {
 			if _, err = tb.Write(report); err != nil {
-				logger.Info("Error writing to telemetry socket", zap.Error(err))
+				tb.logger.Info("Error writing to telemetry socket", zap.Error(err))
```
Comment (tamilmani1989) [inline telemetry/telemetry.go:152]: Logger.Error?

File: telemetry/telemetrybuffer.go [hunk 5]
```diff
@@ -179,10 +202,18 @@ func (tb *TelemetryBuffer) PushData(ctx context.Context) {
...
+			if tb.logger != nil {
+				tb.logger.Info("[Telemetry] server cancel event")
+			} else {
+				log.Logf("[Telemetry] server cancel event")
+			}
 			return
 		case <-ctx.Done():
-			log.Logf("[Telemetry] received context done event")
+			if tb.logger != nil {
+				tb.logger.Info("[Telemetry] received context done event")
```
Comment (vipul-21) [inline telemetry/telemetrybuffer.go:213]: Any reason why we are adding `[Telemetry]` prefix to this info log ? Let's keep it consistent with other info logs for telemetry ? I assume component should tell us now that this is a telemetry log ?

File: telemetry/telemetry.go [hunk 6]
```diff
@@ -112,7 +114,11 @@ func (reportMgr *ReportManager) SendReport(tb *TelemetryBuffer) error {
 		report, err = reportMgr.ReportToBytes()
 		if err == nil {
 			if _, err = tb.Write(report); err != nil {
-				log.Printf("telemetry write failed:%v", err)
+				if reportMgr.Logger != nil {
```
Comment (vipul-21) [inline telemetry/telemetry.go:117]: If this is the only place report manager is logging using zap logger, should we use `tb.logger` here as well ?
In that case we remove the `Logger` from `ReportManager` struct ?

File: telemetry/telemetry.go [hunk 7]
```diff
@@ -143,7 +149,7 @@ func SendCNIMetric(cniMetric *AIMetric, tb *TelemetryBuffer) error {
 		report, err = reportMgr.ReportToBytes()
 		if err == nil {
 			if _, err = tb.Write(report); err != nil {
-				log.Printf("Error writing to telemetry socket:%v", err)
+				tb.logger.Info("Error writing to telemetry socket", zap.Error(err))
```
Comment (vipul-21) [inline telemetry/telemetry.go:152]: `logger.Error`

File: telemetry/telemetrybuffer_test.go [hunk 8]
```diff
@@ -71,7 +71,7 @@ func TestWrite(t *testing.T) {
 	_, closeTBServer := createTBServer(t)
 	defer closeTBServer()
 
-	tbClient := NewTelemetryBuffer()
+	tbClient := NewTelemetryBuffer(nil)
```
Comment (tamilmani1989) [inline telemetry/telemetrybuffer_test.go:74]: one test case with logger ?

**Review by tamilmani1989 — APPROVED** [review]
approved with minor suggestion

File: telemetry/telemetrybuffer_test.go (hunk 8 above)
Comment (paulyufan2) [reply telemetry/telemetrybuffer_test.go:74]: yes I agree we should have one test case with this logger, let me change this

**Review by vipul-21 — DISMISSED** [review]

**Review by tamilmani1989 — DISMISSED** [review]

**Review by vipul-21 — APPROVED** [review]

**Review by tamilmani1989 — APPROVED** [review]

**Pull request merged**